patches/tModLoader/Terraria.ID/ProjectileID.cs.patch

```diff
@@ -0,0 +1,41 @@
+--- src/Terraria\Terraria.ID\ProjectileID.cs
++++ src/tModLoader\Terraria.ID\ProjectileID.cs
+@@ -1,3 +_,4 @@
++using ReLogic.Reflection;
+ using System;
+ 
+ namespace Terraria.ID
+@@ -1199,5 +_,32 @@
+ 		public const short BookStaffShot = 712;
+ 		public const short DD2Win = 713;
+ 		public const short Count = 714;
++
++		public static readonly IdDictionary Search = IdDictionary.Create<ProjectileID, short>();
```

Comment:
In terms of naming I much prefer:
`ProjectileID.Find.Name(123)` or `ProjectileID.FindName(123)` 
or the search variants: `ProjectileID.Search.Name(123)` / `ProjectileID.SearchName(123)`
An alternative design is hosting these methods in a specific cache/query object:
`GameCache.GetProjectileName(123)` or `GameCache.Projectile.GetName(123)`
Which would help against polluting the vanilla class.

Reply:
They are "ID" relevant for the singular purpose of identifying types for human readable serialization, since the English names of items and NPCs and stuff are occasionally untrustworthy (even vanilla has duplicates; Seaweed for example).

Reply:
Yeah since we wanted to implement such cache (soon™️) a class like it would exist anyways, hence my suggestion to maybe make such class already 👍

Reply:
Keep in mind classes to cache each field of each entity type will use up a fair chunk of memory.